common/lib/xmodule/xmodule/assets/library_source_block/LibrarySourcedBlockPicker.jsx

```diff
@@ -0,0 +1,217 @@
+/* globals gettext */
+
+import 'whatwg-fetch';
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'underscore';
+import styles from './style.css';
+
+class LibrarySourcedBlockPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      libraries: [],
+      xblocks: [],
+      searchedLibrary: '',
+      libraryLoading: false,
+      xblocksLoading: false,
+      selectedLibrary: undefined,
+      selectedXblocks: new Set(this.props.selectedXblocks),
+    };
+    this.onLibrarySearchInput = this.onLibrarySearchInput.bind(this);
+    this.onXBlockSearchInput = this.onXBlockSearchInput.bind(this);
+    this.onLibrarySelected = this.onLibrarySelected.bind(this);
+    this.onXblockSelected = this.onXblockSelected.bind(this);
+    this.onDeleteClick = this.onDeleteClick.bind(this);
```

Comment:
I'm a bigger fan of `onFoo = () => {}` when defining the method, because it keeps it all in the same line.  Consider this a nit, though, and not blocking.  If you disagree, carry on. :)

Reply:
I think it's okay to keep this format as that would require updating babel to support class field declarations (that syntax sin't supported in studio's version yet), without much gain

Reply:
Ah, ok, it's a versioning issue.  No biggie, thanks!